tests/Middleware/CompressionGzipMiddlewareTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace React\Tests\Http\Middleware;
+
+use Psr\Http\Message\ServerRequestInterface;
+use React\Http\Middleware\CompressionGzipMiddleware;
+use React\Http\Response;
+use React\Http\ServerRequest;
+use React\Promise\FulfilledPromise;
+use React\Promise\Promise;
+use React\Tests\Http\TestCase;
+
+class CompressionGzipMiddlewareTest extends TestCase
+{
+
+    public function testNoAutoCompressWhenNoSpecificHeadersArePresent()
+    {
+        $content = 'Some response';
+
+        $request = new ServerRequest('GET', 'https://example.com/');
+        $response = new Response(200, array(), $content);
+        $next = $this->getNextCallback($request, $response);
+        $middleware = new CompressionGzipMiddleware();
+
+        /** @var FulfilledPromise $result */
+        $result = $middleware($request, $next);
+        $result->done(function ($value) use (&$response) {
```

Comment:
You can't use `done()` here as we're supporting react/promise 1.x. Same for other methods below.